interfaces/builtin/system_packages_doc_test.go

```diff
@@ -0,0 +1,125 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin_test
+
+import (
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/builtin"
+	"github.com/snapcore/snapd/interfaces/mount"
+	"github.com/snapcore/snapd/release"
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/testutil"
+)
+
+type sytemPackagesDocSuite struct {
+	iface        interfaces.Interface
+	coreSlotInfo *snap.SlotInfo
+	coreSlot     *interfaces.ConnectedSlot
+	plugInfo     *snap.PlugInfo
+	plug         *interfaces.ConnectedPlug
+}
+
+var _ = Suite(&sytemPackagesDocSuite{iface: builtin.MustInterface("system-packages-doc")})
+
+const sytemPackagesDocConsumerYaml = `name: consumer
+version: 0
+apps:
+ app:
+  plugs: [system-packages-doc]
+`
+
+const sytemPackagesDocCoreYaml = `name: core
+version: 0
+type: os
+slots:
+  system-packages-doc:
+`
+
+func (s *sytemPackagesDocSuite) SetUpTest(c *C) {
+	s.plug, s.plugInfo = MockConnectedPlug(c, sytemPackagesDocConsumerYaml, nil, "system-packages-doc")
+	s.coreSlot, s.coreSlotInfo = MockConnectedSlot(c, sytemPackagesDocCoreYaml, nil, "system-packages-doc")
+}
+
+func (s *sytemPackagesDocSuite) TearDownTest(c *C) {
+	dirs.SetRootDir("/")
+}
+
+func (s *sytemPackagesDocSuite) TestName(c *C) {
+	c.Assert(s.iface.Name(), Equals, "system-packages-doc")
+}
+
+func (s *sytemPackagesDocSuite) TestSanitizeSlot(c *C) {
+	c.Assert(interfaces.BeforePrepareSlot(s.iface, s.coreSlotInfo), IsNil)
+}
+
+func (s *sytemPackagesDocSuite) TestSanitizePlug(c *C) {
+	c.Assert(interfaces.BeforePreparePlug(s.iface, s.plugInfo), IsNil)
+}
+
+func (s *sytemPackagesDocSuite) TestAppArmorSpec(c *C) {
+	restore := release.MockOnClassic(true)
```

Comment:
Since we have a release.MockOnClassic(true) we should really have a false that all of this is not there. Not a blocker (ideally in a follow-up).

Reply:
Yup, I'll do this in a follow-up.